src/presign/participant.rs

```diff
@@ -1141,7 +1157,10 @@ impl PresignKeyShareAndInfo {
                     self.aux_info_public.pk(),
                     &Gamma,
                 ),
-                ProverSecret::new(&sender_r1_priv.k, &sender_r1_priv.rho),
+                ProverSecret::new(
+                    &sender_r1_priv.k.get_bignumber_secret().clone(),
```

Comment:
No need to call clone here. `get_bignumber_secret` is already returning the right type here.